src/main/drivers/bus_spi_hal_ll.c

```diff
@@ -189,6 +189,11 @@ bool spiInitDevice(SPIDevice device, bool leadingEdge)
         IOConfigGPIOAF(IOGetByTag(spi->sck), SPI_IO_AF_SCK_CFG_HIGH, spi->af);
     }
     IOConfigGPIOAF(IOGetByTag(spi->miso), SPI_IO_AF_MISO_CFG, spi->af);
+
+    // If MOSI pin for SPI3 is PB2 -> AF7, not AF6
+    if (spiDeviceByInstance(spi->dev) == SPIDEV_3 && IO_TAG(SPI3_MOSI_PIN) == IO_TAG(PB2)) {
```

Comment:
Can't we do it in a different way that is driven by a target file? We should avoid hacks like this as long as possible

Reply:
I think it can be, in this case target file will contain additional defines (any combination of MOSI_AF, MISO_AF and SCK_AF) and if they are defined same function spiInitDevice() will use defined, if not - default one.
Or... we need to implement resource tables like in EmuFlight.

Reply:
@DzikuVx 
Reworked.
Now custom SPI AF's can be taken from target.h if provided. In case they are not - "old" defaults are used (for compatibility).
If custom definintions are used, all three (SCK, MISO, MOSI) should be provided together, otherwise preprocessor error will appear.